tests/evaluate/test_evaluation.py

```diff
@@ -2139,3 +2139,31 @@ def test_import_evaluation_dataset():
     # This test is to validate both imports work at the same time
     from mlflow.models.evaluation import EvaluationDataset
     from mlflow.models.evaluation.base import EvaluationDataset  # noqa: F401
+
+
+def test_evaluate_show_error_stack(linear_regressor_model_uri, diabetes_dataset):
+    with mlflow.start_run():
+        test_err = "Scoring server output 13234"
+        server_proc = mock.MagicMock()
+        with mock.patch(
+            "mlflow.pyfunc.scoring_server.client.ScoringServerClient.wait_server_ready",
+            side_effect=RuntimeError("Wait scoring server ready timeout."),
+        ) as mock_wait_server_ready, mock.patch(
+            "mlflow.pyfunc.backend.PyFuncBackend.serve",
+            return_value=server_proc,
+        ) as mock_serve:
+            server_proc.poll.return_value = None
+            server_proc.poll.return_value = None
+            server_proc.communicate.return_value = (test_err, "")
+            with pytest.raises(Exception, match=test_err):
```

Comment:
```suggestion
            with pytest.raises(MlflowException, match=test_err):
```

`MlflowException` should be raised.